src/tools/miri/src/machine.rs

```diff
@@ -1225,6 +1226,46 @@ impl<'tcx> Machine<'tcx> for MiriMachine<'tcx> {
         })
     }
 
+    /// Called to adjust global allocations to the Provenance and AllocExtra of this machine.
+    ///
+    /// If `alloc` contains pointers, then they are all pointing to globals.
+    ///
+    /// This should avoid copying if no work has to be done! If this returns an owned
+    /// allocation (because a copy had to be done to adjust things), machine memory will
+    /// cache the result. (This relies on `AllocMap::get_or` being able to add the
+    /// owned allocation to the map even when the map is shared.)
+    fn adjust_global_allocation<'b>(
+        ecx: &InterpCx<'tcx, Self>,
+        id: AllocId,
+        alloc: &'b Allocation,
+    ) -> InterpResult<'tcx, Cow<'b, Allocation<Self::Provenance, Self::AllocExtra, Self::Bytes>>>
+    {
+        // The default implementation does a copy; CTFE machines have a more efficient implementation
+        // based on their particular choice for `Provenance`, `AllocExtra`, and `Bytes`.
+        let kind = Self::GLOBAL_KIND
+            .expect("if GLOBAL_KIND is None, adjust_global_allocation must be overwritten");
+        // This closure is special in that it relies on `take_prepared_alloc_bytes` to give it some
+        // reserved memory space in case Miri needs to handle addresses especially for native calls.
+        let alloc = alloc.adjust_from_tcx(&ecx.tcx, |bytes, align| {
+                Ok(if ecx.machine.native_lib.is_some() {
+                    assert_eq!(alloc.align, align);
```

Comment:
What I meant with sanity-checking `align` is checking that `alloc_bytes.as_ptr().addr()` is aligned to `align`.

Reply:
Is
```rust
assert_eq!(prepared_alloc_bytes.as_ptr().align_offset(align.bytes_usize()), 0);
```
the correct approach?

Reply:
Yeah using `is_aligned_to` sounds good!